app/src/ui/preferences/appearance.tsx

```diff
@@ -30,13 +50,51 @@ export class Appearance extends React.Component<IAppearanceProps, {}> {
     } else {
       fatalError(`Unknown theme index ${index}`)
     }
+    this.setState({ automaticallySwitchTheme: false })
+    this.props.onAutomaticallySwitchThemeChanged(false)
+  }
+
+  private onAutomaticallySwitchThemeChanged = (
+    event: React.FormEvent<HTMLInputElement>
+  ) => {
+    const value = event.currentTarget.checked
+
+    if (value) {
+      const usingDarkMode = isDarkModeEnabled()
+      this.onSelectedThemeChanged(usingDarkMode ? 1 : 0)
+    }
+
+    this.setState({ automaticallySwitchTheme: value })
+    this.props.onAutomaticallySwitchThemeChanged(value)
   }
 
   public render() {
     const selectedIndex =
       this.props.selectedTheme === ApplicationTheme.Dark ? 1 : 0
+    const doesSupportDarkMode = supportsDarkMode()
 
-    return (
+    return doesSupportDarkMode ? (
```

Comment:
Rather than repeating the theme selection, can we trying decomposing the main `render` into smaller 
 functions which may return `null` to signal to React that nothing needs to be rendered:

```tsx
public render() {
  return (
    <DialogContent>
      {this.renderThemeOptions()}
      {this.renderAutoSwitcherOption()}
    </DialogContent>
  )
}
```

Where each of these functions is focused on a part of the overall JSX:

```jsx
private renderThemeOptions(): JSX.Element {
  // 
}

private renderAutoSwitcherOption(): JSX.Element | null {
  const isDarkModeSupported = supportsDarkMode()
  if (!isDarkModeSupported) {
    return null
  }

  //
}